src/main/java/net/dv8tion/jda/annotations/ForRemoval.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2015-2018 Austin Keener & Michael Ritter & Florian Spieß
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.annotations;
+
+import java.lang.annotation.*;
+
+/**
+ * Functionality annotated with ForRemoval will no longer be supported
+ * and should not be used anymore in new code.
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
```

Comment:
I'm pretty sure we don't need them at Runtime... For most of these, SOURCE would probably be enough, and if not, CLASS. I don't see why those would be needed at runtime for eg reflection

Reply:
You could write runtime inspections this way

Reply:
I would agree with @kantenkugel here.
If these annotations are really for "documentation" they do not need to be available at runtime via reflection.